src/reflect/type.go

```diff
@@ -611,49 +645,56 @@ func (t *rawType) rawField(n int) rawStructField {
 	flagsByte := *(*byte)(data)
 	data = unsafe.Add(data, 1)
 
-	// Read the field name.
-	nameStart := data
-	var nameLen uintptr
-	for *(*byte)(data) != 0 {
-		nameLen++
-		data = unsafe.Add(data, 1) // C: data++
-	}
-	name := *(*string)(unsafe.Pointer(&stringHeader{
-		data: nameStart,
-		len:  nameLen,
-	}))
+	name := readStringZ(data)
+	data = unsafe.Add(data, len(name))
 
-	// Read the field tag, if there is one.
-	var tag string
-	if flagsByte&structFieldFlagHasTag != 0 {
-		data = unsafe.Add(data, 1) // C: data+1
-		tagLen := uintptr(*(*byte)(data))
-		data = unsafe.Add(data, 1) // C: data+1
-		tag = *(*string)(unsafe.Pointer(&stringHeader{
-			data: data,
-			len:  tagLen,
-		}))
-	}
+	return rawStructFieldFromPointer(field.fieldType, data, flagsByte, name, offset)
+}
 
-	// Set the PkgPath to some (arbitrary) value if the package path is not
-	// exported.
-	pkgPath := ""
-	if flagsByte&structFieldFlagIsExported == 0 {
-		// This field is unexported.
-		// TODO: list the real package path here. Storing it should not
-		// significantly impact binary size as there is only a limited
-		// number of packages in any program.
-		pkgPath = "<unimplemented>"
+// rawFieldByName returns nearly the same value as FieldByName but without converting the
+// Type member to an interface.
+//
+// For internal use only.
```

Comment:
"For internal use only" is already implied by the fact that the method is not exported.